cairo/tests/ethereum/cancun/test_state.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+from ethereum_types.bytes import Bytes32
+from hypothesis import given
+from hypothesis import strategies as st
+
+from ethereum.cancun.fork_types import Address
+from ethereum.cancun.state import State, get_account, get_account_optional, get_storage
+from tests.utils.strategies import address, bytes32, state
+
+pytestmark = pytest.mark.python_vm
+
+
+class TestState:
+    @given(state=state, address=address, draw_address_from_state=st.booleans())
+    def test_get_account(
+        self, cairo_run, state: State, address: Address, draw_address_from_state: bool
+    ):
+        if draw_address_from_state and state._main_trie._data.keys():
+            # drawing the first address from the state's main trie maintains randomness
+            # since state addresses are drawn randomly as part of `state` strategy
+            address = next(iter(state._main_trie._data))
+        [state_cairo, result_cairo] = cairo_run("get_account", state, address)
+        result_py = get_account(state, address)
+        assert result_cairo == result_py
+        assert state_cairo == state
+
+    @given(state=state, address=address, draw_address_from_state=st.booleans())
+    def test_get_account_optional(
+        self, cairo_run, state: State, address: Address, draw_address_from_state: bool
+    ):
+        if draw_address_from_state and state._main_trie._data.keys():
+            address = next(iter(state._main_trie._data))
```

Comment:
```suggestion
    @given(state=state, address=address, draw_address_from_state=...)
    def test_get_account_optional(
        self, cairo_run, state: State, address: Address, draw_address_from_state: bool
    ):
        if draw_address_from_state and len(state._main_trie._data) != 0:
            address = next(iter(state._main_trie._data))
```

Reply:
same here: 
won't this be clearer if divided in 2 tests ?
Something like
`test_get_account_optional_present_in_state`
`test_get_account_optional_absent_in_state`